### PR TITLE
WIP: Resample

### LINKED ISSUE #302: dask.dataframe.resample
Given a series that is indexed and partitioned on a datetime, implement `resample` from pandas.  

If the resample period fits nicely within the partitions then this should be a straightforward operation.  This might be enough to start with.

If we want to get adventurous we could also take a bit from the surrounding partitions similar to how dask.array does ghosting (though far simpler).  If we go this route then it might be good to generalize the overlap logic.

If a continuum person works on this then the anaconda-download-data might be a good test dataset.  I've been playing with it for other projects and can probably provide a decent testbed.

CC @cpcloud this might be up your alley.

### PR DESCRIPTION
**Pull request by cpcloud**
closes #302 

This is very much a work in progress. There are a number of issues that need to be resolved before merging:

- [ ] The default scheduler both fails comparison and segfaults while the single
      threaded scheduler passes tests
- [ ] Standard group by a column is broken in a few places
- [ ] I have to call `fillna(0)` in the aggregation step of count.
      Not sure I can get around this in a clean way.

### CODE REVIEW

File: dask/dataframe/core.py [hunk 1]
```diff
@@ -727,38 +736,41 @@ def apply(func, columns=None):
...
+        chunk = lambda df, index: df.groupby(index).min()
+        agg = lambda df: df.groupby(self.index).min()
         return aca([self.frame, self.index],
                    chunk=chunk, aggregate=agg, columns=[self.key])
 
     def max(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].max()
-        agg = lambda df: df.groupby(level=0).max()
+        chunk = lambda df, index: df.groupby(index).max()
+        agg = lambda df: df.groupby(self.index).max()
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Why these changes?

Comment (mrocklin) [inline dask/dataframe/core.py]: This might fail in cases like 

    g = df.groupby(df.x)

Because `g.index` is no longer a happy string, but rather a `dd.Series` object.

Comment (mrocklin) [inline dask/dataframe/core.py]: (I should add docstrings to the `GroupBy` objects describing their attributes)

Comment (cpcloud) [inline dask/dataframe/core.py]: > Why these changes?

The easiest way to implement `resample` in dask is to use the `pd.TimeGrouper` object. Currently, only a column is allowed to be the grouper, but to allow us to use `pd.TimeGrouper` we need to allow other kinds of objects. This is a WIP, but I wanted to get feedback on the approach before going further.

File: dask/dataframe/core.py [hunk 2]
```diff
@@ -727,38 +736,41 @@ def apply(func, columns=None):
...
+        chunk = lambda df, index: df.groupby(index).sum()
+        agg = lambda df: df.groupby(self.index).sum()
         return aca([self.frame, self.index],
                    chunk=chunk, aggregate=agg, columns=[self.key])
 
     def min(self):
-        chunk = lambda df, index: df.groupby(index)[self.key].min()
-        agg = lambda df: df.groupby(level=0).min()
+        chunk = lambda df, index: df.groupby(index).min()
+        agg = lambda df: df.groupby(self.index).min()
```
Comment (mrocklin) [inline dask/dataframe/core.py]: Why did you end up needing this.  I'm running into an issue with `groupby(...).apply` that has me confused.
